include/seqan3/alignment/pairwise/align_result_selector.hpp

```diff
@@ -116,8 +120,39 @@ struct align_result_selector
         }
     }
 
-    //!\brief The determined result type.
-    using type = decltype(_determine());
+    //!\brief Augments the result type with the debug output for score and trace matrix.
+    template <typename alignment_result_value_t>
+    static constexpr auto augment_if_debug(alignment_result_value_t)
+    {
+        if constexpr (!configuration_t::template exists<align_cfg::debug_type>())
```

Comment:
In general, it is better style to not negate the condition and instead just switch the code blocks, but that's :nail_care: